crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation/da_cost_per_byte.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{
+    pretty,
+    simulation::gen_noisy_signal,
+    Source,
+};
+use std::iter;
+
+pub fn get_da_cost_per_byte_from_source(
+    source: Source,
+    update_period: usize,
+) -> Vec<u64> {
+    match source {
+        Source::Generated { size } => arbitrary_cost_per_byte(size, update_period),
+        Source::Predefined {
+            file_path,
+            sample_size,
+        } => {
+            let original = get_costs_from_csv_file(&file_path, sample_size);
+            original
+                .into_iter()
+                .map(|x| std::iter::repeat(x).take(update_period))
+                .flatten()
+                .collect()
+        }
+    }
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct Record {
+    block_number: u64,
+    excess_blob_gas: u64,
+    blob_gas_used: u64,
+    blob_fee_wei: u64,
+    blob_fee_wei_for_1_blob: u64,
+    blob_fee_wei_for_2_blobs: u64,
+    blob_fee_wei_for_3_blobs: u64,
+}
+
+fn get_costs_from_csv_file(file_path: &str, sample_size: Option<usize>) -> Vec<u64> {
+    let mut rdr = csv::ReaderBuilder::new()
+        .has_headers(true)
+        .from_path(file_path)
+        .unwrap();
+    let mut costs = vec![];
+    let headers = csv::StringRecord::from(vec![
+        "block_number",
+        "excess_blob_gas",
+        "blob_gas_used",
+        "blob_fee_wei",
+        "blob_fee_wei_for_1_blob",
+        "blob_fee_wei_for_2_blobs",
+        "blob_fee_wei_for_3_blobs",
+    ]);
+    let mut max_cost = 0;
+    for record in rdr.records().skip(1) {
+        if let Some(size) = sample_size {
+            if costs.len() >= size {
+                break;
+            }
+        };
+        let record: Record = record.unwrap().deserialize(Some(&headers)).unwrap();
+        let cost = record.blob_fee_wei;
+        if cost > max_cost {
+            max_cost = cost;
+        }
+        costs.push(cost);
+    }
+    println!("Max cost: {}", pretty(max_cost));
+    costs
+}
+
+fn noisy_eth_price<T: TryInto<f64>>(input: T) -> f64
+where
+    <T as TryInto<f64>>::Error: core::fmt::Debug,
```

Comment:
curious, is this constraint required?

Reply:
Which part. I'm using it for more than one `From` type, hence the `TryInto`, and the `Debug` is for the `unwrap` :P

<img width="715" alt="image" src="https://github.com/user-attachments/assets/0f42dd92-68c6-4b61-9fc5-2b6ebbeadb51">


I guess we're moving away from `unwrap` in the charts. Maybe worth doing it for the simulation code too.

Reply:
eh, yeah ~ feel free to ignore